### ISSUE TITLE
RBAC Error to reconciler controller.argo

### ISSUE THREAD

**Issue by CharcoGreen** [redhat-developer/gitops-operator#518]
**Describe the bug**
I have a problem with the gitops operator in openshift. The fact is that it is not capable of raising argocd instances, it seems like a permissions problem or when applying rbac. 

An error can be observed in the operator related to the controller and the reconciliation

**To Reproduce**
I install the Red Hat OpenShift GitOps operator from the CLI or from the GUI. when i want to see the objects of the argo instance, only kam and cluster appear

```
oc get po  -n openshift-gitops
NAME                       READY   STATUS    RESTARTS   AGE
cluster-84755c76c6-n552c   1/1     Running   0          42m
kam-946f49d74-466xq        1/1     Running   0          42m
```

Argo operator logs.

```
1.685015672578439e+09   INFO    controller.argocd       Reconciling ArgoCD      {"reconciler group": "argoproj.io", "reconciler kind": "ArgoCD", "name": "openshift-gitops", "namespace": "openshift-gitops", "namespace": "openshift-gitops", "name": "openshift-gitops"}
1.6850156725786343e+09  INFO    controller_argocd       reconciling status
1.6850156725787117e+09  INFO    controller_argocd       argocd-server route requested but not found on cluster
1.6850156725787256e+09  INFO    controller_argocd       reconciling roles
1.6850156725787392e+09  INFO    openshift_controller_argocd     configuring policy rule for Application Controller      {"ArgoCD Namespace": "openshift-gitops", "ArgoCD Name": "openshift-gitops"}
1.6850156726094017e+09  ERROR   controller.argocd       Reconciler error        {"reconciler group": "argoproj.io", "reconciler kind": "ArgoCD", "name": "openshift-gitops", "namespace": "openshift-gitops", "error": "Role.rbac.authorization.k8s.io \"openshift-gitops-argocd-application-controller\" is invalid: rules[18].nonResourceURLs: Invalid value: []string{\"*\"}: namespaced rules cannot apply to non-resource URLs"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /remote-source/deps/gomod/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
        /remote-source/deps/gomod/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227
1.685015677729543e+09   INFO    controller.argocd       Reconciling ArgoCD      {"reconciler group": "argoproj.io", "reconciler kind": "ArgoCD", "name": "openshift-gitops", "namespace": "openshift-gitops", "namespace": "openshift-gitops", "name": "openshift-gitops"}
1.6850156777297578e+09  INFO    controller_argocd       reconciling status
1.6850156777298887e+09  INFO    controller_argocd       argocd-server route requested but not found on cluster
1.685015677729913e+09   INFO    controller_argocd       reconciling roles
1.685015677729941e+09   INFO    openshift_controller_argocd     configuring policy rule for Application Controller      {"ArgoCD Namespace": "openshift-gitops", "ArgoCD Name": "openshift-gitops"}
1.6850156777524261e+09  ERROR   controller.argocd       Reconciler error        {"reconciler group": "argoproj.io", "reconciler kind": "ArgoCD", "name": "openshift-gitops", "namespace": "openshift-gitops", "error": "Role.rbac.authorization.k8s.io \"openshift-gitops-argocd-application-controller\" is invalid: rules[18].nonResourceURLs: Invalid value: []string{\"*\"}: namespaced rules cannot apply to non-resource URLs"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /remote-source/deps/gomod/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
        /remote-source/deps/gomod/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227
```

**Expected behavior**
Deploy the argocd instances

**Screenshots**

**Additional context**

**Version de Red Hat OpenShift GitOps:** Stable 1.5.10
**Openshift Version** 4.10.56

```
apiVersion: argoproj.io/v1alpha1
kind: ArgoCD
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"argoproj.io/v1alpha1","kind":"ArgoCD","metadata":{"annotations":{},"finalizers":["argoproj.io/finalizer"],"generation":2,"name":"openshift-gitops","namespace":"openshift-gitops"},"spec":{"applicationSet":{"resources":{"limits":{"cpu":"2","memory":"1Gi"},"requests":{"cpu":"250m","memory":"512Mi"}}},"controller":{"processors":{},"resources":{"limits":{"cpu":"2","memory":"2Gi"},"requests":{"cpu":"250m","memory":"1Gi"}},"sharding":{}},"dex":{"openShiftOAuth":true,"resources":{"limits":{"cpu":"500m","memory":"256Mi"},"requests":{"cpu":"250m","memory":"128Mi"}}},"grafana":{"enabled":false,"ingress":{"enabled":false},"resources":{"limits":{"cpu":"500m","memory":"256Mi"},"requests":{"cpu":"250m","memory":"128Mi"}},"route":{"enabled":false}},"ha":{"enabled":false,"resources":{"limits":{"cpu":"500m","memory":"256Mi"},"requests":{"cpu":"250m","memory":"128Mi"}}},"initialSSHKnownHosts":{},"prometheus":{"enabled":false,"ingress":{"enabled":false},"route":{"enabled":false}},"rbac":{"policy":null,"scopes":"[groups]"},"redis":{"resources":{"limits":{"cpu":"500m","memory":"256Mi"},"requests":{"cpu":"250m","memory":"128Mi"}}},"repo":{"resources":{"limits":{"cpu":"1","memory":"1Gi"},"requests":{"cpu":"250m","memory":"256Mi"}}},"resourceExclusions":"- apiGroups:\n  - tekton.dev\n  clusters:\n  - '*'\n  kinds:\n  - TaskRun\n  - PipelineRun\n","server":{"autoscale":{"enabled":false},"grpc":{"ingress":{"enabled":false}},"ingress":{"enabled":false},"resources":{"limits":{"cpu":"500m","memory":"256Mi"},"requests":{"cpu":"125m","memory":"128Mi"}},"route":{"enabled":true},"service":{"type":""}},"tls":{"ca":{}}}}
  creationTimestamp: "2023-05-25T11:25:07Z"
  finalizers:
  - argoproj.io/finalizer
  generation: 1
  name: openshift-gitops
  namespace: openshift-gitops
  resourceVersion: "890415464"
  uid: fcb39881-b7d9-48c2-bbaa-5e06eccff55b
spec:
  applicationSet:
    resources:
      limits:
        cpu: "2"
        memory: 1Gi
      requests:
        cpu: 250m
        memory: 512Mi
  controller:
    processors: {}
    resources:
      limits:
        cpu: "2"
        memory: 2Gi
      requests:
        cpu: 250m
        memory: 1Gi
    sharding: {}
  dex:
    openShiftOAuth: true
    resources:
      limits:
        cpu: 500m
        memory: 256Mi
      requests:
        cpu: 250m
        memory: 128Mi
  grafana:
    enabled: false
    ingress:
      enabled: false
    resources:
      limits:
        cpu: 500m
        memory: 256Mi
      requests:
        cpu: 250m
        memory: 128Mi
    route:
      enabled: false
  ha:
    enabled: false
    resources:
      limits:
        cpu: 500m
        memory: 256Mi
      requests:
        cpu: 250m
        memory: 128Mi
  initialSSHKnownHosts: {}
  prometheus:
    enabled: false
    ingress:
      enabled: false
    route:
      enabled: false
  rbac:
    scopes: '[groups]'
  redis:
    resources:
      limits:
        cpu: 500m
        memory: 256Mi
      requests:
        cpu: 250m
        memory: 128Mi
  repo:
    resources:
      limits:
        cpu: "1"
        memory: 1Gi
      requests:
        cpu: 250m
        memory: 256Mi
  resourceExclusions: |
    - apiGroups:
      - tekton.dev
      clusters:
      - '*'
      kinds:
      - TaskRun
      - PipelineRun
  server:
    autoscale:
      enabled: false
    grpc:
      ingress:
        enabled: false
    ingress:
      enabled: false
    resources:
      limits:
        cpu: 500m
        memory: 256Mi
      requests:
        cpu: 125m
        memory: 128Mi
    route:
      enabled: true
    service:
      type: ""
  tls:
    ca: {}
status:
  applicationController: Unknown
  dex: Unknown
  phase: Pending
  redis: Unknown
  repo: Unknown
  server: Unknown
  ssoConfig: Success
```

Can you help me?

**Comment by anandf** [comment]
@CharcoGreen Could you try using the `gitops-1.8` channel instead of the `stable` channel. The `stable` channel is deprecated. `gitops-1.8` is supported from on OCP versions 4.10-4.13 (https://docs.openshift.com/container-platform/4.10/cicd/gitops/gitops-release-notes.html)

**Comment by CharcoGreen** [comment]
Hello,

I have solved the problem.

We had tried everything, update to 1.8, reinstall, clean everything etc.

But finally it was a problem in a clusterrole, someone has created a cluster role from a copy paste from the clusterrole edit . Therefore the clusterrole admin (immutable) was getting updates (resources) from this clusterrole.

flow: copiedClusterRole > editClusterRole > adminClusterRole.

Therefore we have modified this clusterrole so that it does not propagate resources.

This issue has been fixed.

Thank you.
